Main.java (Simulação e Exceções)

package banco;

public class Main {
    public static void main(String[] args) {
        ContaCorrente conta1 = new ContaCorrente("João", 1500);
        ContaPoupanca conta2 = new ContaPoupanca("Maria", 800);

        try {
            System.out.println("Saldo do João: R$ " + conta1.getSaldo());
            System.out.println("Saldo da Maria: R$ " + conta2.getSaldo());

            conta1.transferir(conta2, 300);
            System.out.println("Transferência realizada!");

            System.out.println("Saldo do João: R$ " + conta1.getSaldo());
            System.out.println("Saldo da Maria: R$ " + conta2.getSaldo());

            conta2.sacar(1500); // vai causar erro!

        } catch (Exception e) {
            System.out.println("Erro ao realizar operação: " + e.getMessage());
        }

        System.out.println("Fim das operações.");
    }
}
